### PR TITLE
Expand Battlefield News when a new news post has been downloaded

### LINKED ISSUE #7006: Battlefield News not read
Despite the fact that the upcoming tournament is announced in the Battlefield News, few people are aware of it. It means that only few people "open" the Battlefield News. Could you show it opened by default ?

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/6524849-battlefield-news-not-read?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by penev92**
Supersedes #8140.
Closes #7006.

### CODE REVIEW

File: OpenRA.Mods.Common/Widgets/Logic/MainMenuLogic.cs [hunk 1]
```diff
@@ -200,21 +198,17 @@ public MainMenuLogic(Widget widget, World world)
...
 				var newsButton = newsBG.GetOrNull<DropDownButtonWidget>("NEWS_BUTTON");
+
 				if (newsButton != null)
 				{
-					newsButton.OnClick = () =>
-					{
-						newsButton.AttachPanel(newsPanel);
-						newsHighlighted = false;
-					};
+					// Only query for new stories once per day
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/Widgets/Logic/MainMenuLogic.cs]: `s/new stores/news`, also comments end with a full-stop.

Comment (penev92) [inline OpenRA.Mods.Common/Widgets/Logic/MainMenuLogic.cs]: Fixed.

File: OpenRA.Mods.Common/Widgets/Logic/MainMenuLogic.cs [hunk 2]
```diff
@@ -200,21 +198,17 @@ public MainMenuLogic(Widget widget, World world)
...
+
 				if (newsButton != null)
 				{
-					newsButton.OnClick = () =>
-					{
-						newsButton.AttachPanel(newsPanel);
-						newsHighlighted = false;
-					};
+					// Only query for news once per day.
+					var cacheValid = currentNews != null && DateTime.Today.ToUniversalTime() <= Game.Settings.Game.NewsFetchedDate;
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Widgets/Logic/MainMenuLogic.cs]: While you are at it you can make use of `DateTime.UtcNow` instead. http://blogs.msdn.com/b/kirillosenkov/archive/2012/01/10/datetime-utcnow-is-generally-preferable-to-datetime-now.aspx

**Comment by Mailaender** [comment]
Works as promised. :white_check_mark:

**Comment by obrakmann** [comment]
Looks fine to me, too. :+1:

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/00fac00fc0586ad73f6a0f314ae189c02aff77fd%5E...00fac00fc0586ad73f6a0f314ae189c02aff77fd)

**Pull request merged**